### DISCUSSION TITLE
Xcode errors in tutorial

### DISCUSSION THREAD

**Discussion by mtancock** [pointfreeco/swift-composable-architecture#3378] [General]
I'm working through the first SyncUps tutorial and after getting to the end of the section, finding that Xcode is reporting _a lot_ of errors. 
![CleanShot 2024-09-12 at 15 08 20@2x](https://github.com/user-attachments/assets/24a03632-d542-4f95-a332-bb70e675a79c)

It does preview (once I'd fixed the actual errors—tracking down the real errors with all of this was almost impossible).

Any idea if it's possible to get Xcode to just display legit errors (I'm using Xcode 15.4) or if these will eventually go away? Cleaning / restarting Xcode doesn't seem to have helped.

**Comment by kamcma** [comment]
Unfortunately I have seen lots of reports of that `macro implementation ... could not be found` error. I think it is just Xcode bugginess. I suspect most of your errors are simply flowing from that, because without the root `@Reducer` macro working, your reducers are not necessarily in compiling order. I have not seen a certain fix, you just have to try the usual suspects: restart, delete derived data, etc.

You could also try installing Xcode 16, which is release candidate now, and should be officially released Monday.
